app/src/main/java/com/trianguloy/urlchecker/modules/companions/DenylistOption.java

```diff
@@ -0,0 +1,50 @@
+package com.trianguloy.urlchecker.modules.companions;
+
+import com.trianguloy.urlchecker.R;
+import com.trianguloy.urlchecker.utilities.Enums;
+
+/**
+ * Enum representing the app that can be hidden from "Open with" and "Share via".
+
+ * Each entry has:
+ * An id (for saving as a preference),
+ * A label (string resource for UI),
+ * The actual package name of the app to filter out.
+
+ */
+
+public enum DenylistOption implements Enums.IdEnum, Enums.StringEnum {
```

Comment:
Even though this is a good first implementation, I wonder if it's too restrictive.
You can only choose from a predefined list, and you can't choose more than one.

I prefer to store a multilinestring, where each line is one package (empty or invalid lines ignored). This will either require implement a chooser dialog dynamically loaded with the apps...or just leave the textarea and let the user write the packages (not user friendly, I know, but more powerful).

Or, even better, with something similar to the 'templates' in the webhook module. Yes, I think this will be easy to implement, and could be easily replaced with a more powerful dialog if needed.